### ISSUE TITLE
Partial residuals not available.

### ISSUE THREAD

**Issue by fezoz** [strengejacke/ggeffects#225]
Hi, 

after udpating to the new version of the package, I can no longer plot partial residuals from mixed effect models using lme/lmer. under the new version I get 

"Partial residuals not available.
Warning message:
Could not extract residuals"

the issue dissapears if I use a simple lm. Any help with this would be much appreciated. would it be wise to re-install the older version of the package to avoid this issue? below is a reproducible example

```
data(efc)

#mixed effect model
mod1 <- lme(barthtot  ~ grp+c12hour  + e42dep, data = efc,
                       random=~1|negc7d,
                       method="REML",
                       na.action=na.omit)
mod.df <- ggemmeans(mod1, terms = c("grp"), partial.residuals=TRUE)
plot(mod.df, residuals = TRUE)

#simple model
mod2 <- lm(barthtot  ~ grp+c12hour  + e42dep, data = efc,
            na.action=na.omit)

mod.df2 <- ggemmeans(mod2, terms = c("grp"), partial.residuals=TRUE)

plot(mod.df2, residuals = TRUE)
```

**Comment by strengejacke** [comment]
I'm not sure if this ever worked for lme, but it works for lme4:

``` r
library(ggeffects)
library(lme4)
#> Loading required package: Matrix
data(efc)

# mixed effect model
mod1 <- lmer(barthtot ~ grp + c12hour + e42dep + (1 | negc7d),
  data = efc
)
mod.df <- ggpredict(mod1, terms = "grp")
plot(mod.df, residuals = TRUE)
```

![](https://i.imgur.com/POOEI0i.png)

<sup>Created on 2023-02-22 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by strengejacke** [comment]
lme has no working residuals, so I think this is something that can be fixed easily (unless response residuals also work for partial residuals - @mattansb any idea?)

**Comment by adrilugue** [comment]
Hello, I'm having the same problem with gls models from the nlme package. Any help would be really appreciated. Thanks

**Comment by strengejacke** [comment]
I think this is the same issue, that there are no working residuals available. I'm not sure if @mattansb or someone else knows whether other residual types can be used in this case?

**Comment by mattansb** [comment]
In linear models, working residuals and response residuals are the same thing:

``` r
m <- lm(mpg ~ hp, mtcars)
residuals(m, type = "working")[1:4]
#>      Mazda RX4  Mazda RX4 Wag     Datsun 710 Hornet 4 Drive 
#>     -1.5937500     -1.5937500     -0.9536307     -1.1937500

residuals(m, type = "response")[1:4]
#>      Mazda RX4  Mazda RX4 Wag     Datsun 710 Hornet 4 Drive 
#>     -1.5937500     -1.5937500     -0.9536307     -1.1937500
```

<sup>Created on 2023-05-09 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>


So you can use response residuals for lme models.

**Comment by strengejacke** [comment]
Thanks, should be fixed in #321

``` r
library(ggeffects)
library(nlme)
data(efc)

#mixed effect model
mod1 <- lme(barthtot  ~ grp+c12hour  + e42dep, data = efc,
                       random=~1|negc7d,
                       method="REML",
                       na.action=na.omit)
mod.df <- ggemmeans(mod1, terms = "grp")
plot(mod.df, residuals = TRUE)
```

![](https://i.imgur.com/56wogT1.png)

<sup>Created on 2023-05-09 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
